tests/test_config_edge_cases.py

```
@@ -1055,91 +1055,176 @@ def test_validate_yaml_syntax_error_with_syntax_issues(self):
+
+        config = {"credentials_path": "/some/path"}
+        with patch("mmrelay.config.logger") as mock_logger:
+            _warn_on_legacy_path_overrides(config)
+            _warn_on_legacy_path_overrides(config)
+            assert mock_logger.warning.call_count == 1
+
+    def test_no_warning_when_no_legacy_overrides(self):
+        from mmrelay.config import _warn_on_legacy_path_overrides
+
+        with patch.dict(os.environ, {}, clear=False):
+            with patch("mmrelay.config.logger") as mock_logger:
```

Ruff (0.15.10)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)